omegaconf/_utils.py

```diff
@@ -30,12 +30,48 @@
     attr = None  # type: ignore # pragma: no cover
 
 
-def isint(s: str) -> bool:
-    try:
-        int(s)
-        return True
-    except ValueError:
-        return False
+# source: https://yaml.org/type/bool.html
+YAML_BOOL_TYPES = [
+    "y",
+    "Y",
+    "yes",
+    "Yes",
+    "YES",
+    "n",
+    "N",
+    "no",
+    "No",
+    "NO",
+    "true",
+    "True",
+    "TRUE",
+    "false",
+    "False",
+    "FALSE",
+    "on",
+    "On",
+    "ON",
+    "off",
+    "Off",
+    "OFF",
+]
+
+
+class OmegaConfDumper(yaml.Dumper):  # type: ignore
+    def str_representer(dumper: yaml.Dumper, data: str) -> yaml.ScalarNode:
+        with_quotes = yaml_is_bool(data) or is_int(data) or is_float(data)
+        return dumper.represent_scalar(
+            yaml.resolver.BaseResolver.DEFAULT_SCALAR_TAG,
+            data,
+            style=("'" if with_quotes else None),
+        )
+
+
+OmegaConfDumper.add_representer(str, OmegaConfDumper.str_representer)
```

Comment:
instead of calling this static function, can you just add it directly on the OmegaConfDumper object?

Reply:
In the pyyaml source code representers are added that way since pyyaml requires a Dumper class not a object.

Reply:
maybe i can define a function that creates that class, adds representer and returns the class

Reply:
If it wants a class, give it a class.
Can you define the class without calling this helper function?
it seems to me like the representers should be a list on the class.
can you add to it directoy?